src/app/test_executive/payments_test.ml

```diff
@@ -416,15 +435,138 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
          let sender = untimed_node_b in
          let%bind sender_pub_key = pub_key_of_node sender in
          let%bind _ =
+           (*
```

Comment:
Is this section needed? You are sending a bunch of transaction after snark key change to emit ledger proofs

Reply:
The way i've structured it is:

1: make sure snarking works when booted up with a certain key, by making sure a ledger proof is emitted and then checking the balance of the snark account.

2: change the snark key

3: make sure snarking still works as expected with the new key,  by making sure another ledger proof is emitted and then checking the balance of the new snark account.

Each time we check to make sure it works, it requires a ledger proof to be fully formed and emitted, which requires a bunch of txns to be sent.

I think the test is most convincing when we can see that `snark-key1` AND `snark-key2` both have above-zero balances, then we know that the snark node works if you start it with a certain key, as well as functioning correctly after the key is switched out.

We could arguably skip step (1) to save time but then hypothetically the snark node could be bugged on initialization but be fixed when the key is switched.

So tldr, I think this section is not redundant, but arguably a little extra.  If you object, I can remove it.

Reply:
>We could arguably skip step (1) to save time but then hypothetically the snark node could be bugged on initialization but be fixed when the key is switched.
You are checking both keys' balances so any issue with the initialization should be caught?

Anyway, I think it should be ok- it seems to be adding around 5mins extra time.

Reply:
if we change the snark key before a proof is emitted, then the initial snark key won't earn anything right? or am i misunderstanding something?

And I think the fact that it's "only" adding around 5 minutes of time may be assisted by the fact that i increased the resources.